marine.c

```diff
@@ -633,7 +634,50 @@ WS_DLL_PUBLIC int validate_display_filter(char *dfilter) {
     return TRUE;
 }
 
-WS_DLL_PUBLIC int marine_add_filter(char *bpf, char *dfilter, char **fields, int fields_len, char *err_msg) {
+int parse_output_fields(output_fields_t *output_fields, char **fields, int fields_len, char **err_msg) {
+    for (int i = 0; i < fields_len; i++) {
+        output_fields_add(output_fields, fields[i]);
+    }
+
+    GSList *it = NULL;
```

Comment:
Im not sure, but can't you create this variable inside the for loop?
like
`for (GSList *it = invalid_fields; it != NULL; it = g_slist_next(it))`
or something?